allspice/utils/bom_generation.py

```diff
@@ -392,26 +331,70 @@ def _extract_components(
     return components
 
 
-def _group_components(components: list[SchematicComponent]) -> Bom:
-    grouped_components = {}
+def _map_attributes(
+    components: list[dict[str, str | None]],
+    columns: dict[str, list[str]],
+) -> list[dict[str, str]]:
+    """
+    Map the attributes of the components to the columns of the BOM using the
+    columns mapping.
+    """
+
+    mapped_components = []
 
     for component in components:
-        key = component.part_number
+        mapped_component = {}
+        for column, attributes in columns.items():
+            mapped_component[column] = str(_find_first_matching_key(attributes, component) or "")
```

Comment:
I don't expect find_first_matching_key to return the value of the key maybe either split them into find key then get the value or get_first_matching_key_value

Reply:
Changed!